modules/userdirectory-canvas/src/main/java/org/opencastproject/userdirectory/canvas/CanvasUserProviderFactory.java

```diff
@@ -0,0 +1,253 @@
+    if (StringUtils.isBlank(token)) throw new ConfigurationException(CANVAS_TOKEN_KEY, "is not set");
+
+    String courseIdentifierProperty = (String) properties.get(COURSE_IDENTIFIER_PROPERTY_KEY);
+    if (!"id".equals(courseIdentifierProperty) && !"sis_course_id".equals(courseIdentifierProperty)) {
+      // Default to using `id`
+      courseIdentifierProperty = "id";
+    }
+
+    String coursePattern = (String) properties.get(COURSE_PATTERN_KEY);
+    String userPattern = (String) properties.get(USER_PATTERN_KEY);
+
+    String userNameProperty = (String) properties.get(USER_NAME_PROPERTY_KEY);
+    if (!"name".equals(userNameProperty) && !"short_name".equals(userNameProperty)) {
+      // Default to using `short_name`
+      userNameProperty = "short_name";
+    }
+
+    int cacheSize = 1000;
+    try {
+      if (properties.get(CACHE_SIZE) != null) {
+        Integer configuredCacheSize = Integer.parseInt(properties.get(CACHE_SIZE).toString());
+        if (configuredCacheSize != null) {
+          cacheSize = configuredCacheSize.intValue();
+        }
+      }
+    } catch (Exception e) {
+      logger.warn("{} could not be loaded, default value is used: {}", CACHE_SIZE, cacheSize);
+    }
+
+
+    int cacheExpiration = 60;
+    try {
+      if (properties.get(CACHE_EXPIRATION) != null) {
+        Integer configuredCacheExpiration = Integer.parseInt(properties.get(CACHE_EXPIRATION).toString());
+        if (configuredCacheExpiration != null) {
+          cacheExpiration = configuredCacheExpiration.intValue();
+        }
+      }
+    } catch (Exception e) {
+      logger.warn("{} could not be loaded, default value is used: {}", CACHE_EXPIRATION, cacheExpiration);
+    }
+
+    // Instructor roles
+    Set<String> instructorRoles;
+    String instructorRoleList = (String) properties.get(CANVAS_INSTRUCTOR_ROLES_KEY);
+
+    if (!StringUtils.isEmpty(instructorRoleList)) {
+      String trimmedRoles = StringUtils.trim(instructorRoleList);
+      String[] roles = trimmedRoles.split(",");
+      instructorRoles = new HashSet<String>(Arrays.asList(roles));
+      logger.debug("Canvas instructor roles: {}", Arrays.toString(roles));
+    } else {
+      // Default instructor roles
+      instructorRoles = new HashSet<String>();
+      instructorRoles.add("TeacherEnrollment");
+    }
+
+    // Now that we have everything we need, go ahead and activate a new provider, removing an old one if necessary
+    ServiceRegistration existingRegistration = providerRegistrations.remove(pid);
+    if (existingRegistration != null) {
+      existingRegistration.unregister();
+    }
+
+    Organization org;
+    try {
+      org = orgDirectory.getOrganization(organization);
+    } catch (NotFoundException e) {
+      logger.warn("Organization {} not found!", organization);
+      throw new ConfigurationException(ORGANIZATION_KEY, "not found");
+    }
+
+    logger.debug("Creating new CanvasUserProviderInstance for pid=" + pid);
```

Comment:
Use:
```suggestion
    logger.debug("Creating new CanvasUserProviderInstance for pid={}", pid);
```
that way you only build the string if the log level really is set to debug or higher. There are more instances of this. In the new code.